### ISSUE TITLE
Type error: Object is possibly 'undefined'

### ISSUE THREAD

**Issue by bravo-kernel** [wojtekmaj/get-user-locale#51]
Any pointers on what might cause below CI error when using v2.1.1:

```
5:38:43 PM: #9 1.141 info  - Linting and checking validity of types...
5:38:55 PM: #9 13.14 Failed to compile.
5:38:55 PM: #9 13.14 
5:38:57 PM: #9 13.14 ./node_modules/get-user-locale/src/index.ts:23:29
5:38:57 PM: #9 13.14 Type error: Object is possibly 'undefined'.
5:38:57 PM: #9 13.14 
5:38:57 PM: #9 13.14   21 | 
5:38:57 PM: #9 13.14   22 |     const splitEl = el.split('-');
5:38:57 PM: #9 13.14 > 23 |     return `${splitEl[0]}-${splitEl[1].toUpperCase()}`;
5:38:57 PM: #9 13.14      |                             ^
5:38:57 PM: #9 13.14   24 |   });
5:38:57 PM: #9 13.14   25 | }
```

**Comment by wojtekmaj** [comment]
Hmmm... The error does make sense to me. `String.split()` returns an array of strings and the second element of this array indeed can be undefined

But:
* My TypeScript does not complain about that - I wonder which TypeScript version are you using?
* Are there any additional dependencies that may have interfered with what the return type of `String.split()` is?

**Comment by bravo-kernel** [comment]
Thanks for responding. I am using Typescript `4.9.4` and will try to follow-up on your questions.

**Comment by bravo-kernel** [comment]
The error appears locally in my project when I run `npx tsc`. If you want I can give you access to my private repo.

**Comment by bravo-kernel** [comment]
I have created a fresh project at https://github.com/bravo-kernel/debug-get-user-locale.

Then added `get-user-locale` line [here](https://github.com/bravo-kernel/debug-get-user-locale/blob/main/src/pages/index.tsx#L12).

If you run `npx tsc` you will see the error.

**Comment by wojtekmaj** [comment]
Aaaah, thank you for this.

The key here is `noUncheckedIndexedAccess`. When turned on, you'll get the error. TypeScript disregards the safety checks I have put in place and throws this error. 

Related: https://github.com/microsoft/TypeScript/issues/41638

**Comment by bravo-kernel** [comment]
Cool, good that we both are seeing the same. Are you suggesting that I turn off `noUncheckedIndexedAccess` (which is not an option)?

**Comment by wojtekmaj** [comment]
Not at all :D Just thinking out loud.

**Comment by wojtekmaj** [comment]
Released in v2.1.2. Thank you for this!

**Comment by bravo-kernel** [comment]
Perfect, much appreciated 💃
